youtube_dl/extractor/common.py

```diff
@@ -806,24 +853,59 @@ def _formats_key(f):
                     ext_preference = -1
                 audio_ext_preference = 0
 
-            return (
-                preference,
-                f.get('language_preference') if f.get('language_preference') is not None else -1,
-                f.get('quality') if f.get('quality') is not None else -1,
-                f.get('tbr') if f.get('tbr') is not None else -1,
-                f.get('filesize') if f.get('filesize') is not None else -1,
-                f.get('vbr') if f.get('vbr') is not None else -1,
-                f.get('height') if f.get('height') is not None else -1,
-                f.get('width') if f.get('width') is not None else -1,
-                proto_preference,
-                ext_preference,
-                f.get('abr') if f.get('abr') is not None else -1,
-                audio_ext_preference,
-                f.get('fps') if f.get('fps') is not None else -1,
-                f.get('filesize_approx') if f.get('filesize_approx') is not None else -1,
-                f.get('source_preference') if f.get('source_preference') is not None else -1,
-                f.get('format_id') if f.get('format_id') is not None else '',
+            def synthetic_criterion(name, value):
+                return name, value
+
+            def field_criterion(name, default=-1):
+                return name, f.get(name) if f.get(name) is not None else default
+
+            default_key = (
+                synthetic_criterion('preference', preference),
+                field_criterion('language_preference'),
+                field_criterion('quality'),
+                field_criterion('tbr'),
+                field_criterion('filesize'),
+                field_criterion('vbr'),
+                field_criterion('height'),
+                field_criterion('width'),
+                synthetic_criterion('proto_preference', proto_preference),
+                synthetic_criterion('ext_preference', ext_preference),
+                field_criterion('abr'),
+                synthetic_criterion('audio_ext_preference', audio_ext_preference),
+                field_criterion('fps'),
+                field_criterion('filesize_approx'),
+                field_criterion('source_preference'),
+                field_criterion('format_id', ''),
             )
+
+            if (not criteria_preference and not criteria_relative_preference and
+                    not criteria_exclusion):
+                return default_key
```

Comment:
You are returning a tuple of `(filed, value)` item instead of a list of values as you do later.